src/transformers/utils/tensor_parallel_utils.py

```diff
@@ -0,0 +1,147 @@
+from .. import BertPreTrainedModel, GPT2PreTrainedModel, TransfoXLPreTrainedModel
+
+
+# ie. nn.Linear(3 * dim, dim)
+# only defined the models that have fused attention layer.
+FUSED_ATTENTION_MAPPING = {
+    GPT2PreTrainedModel: {"attn.c_attn": 3, "crossattention.c_attn": 2},
+    TransfoXLPreTrainedModel: {"qkv_net": 3},
+}
+
+# ie. nn.Linear(out_dim, in_dim) or Conv1D()
+# only defined the models that have reversed parameters.
+REVERSED_PARAM_MAPPING = {
+    GPT2PreTrainedModel: ["attn", "crossattention", "mlp"],
+    TransfoXLPreTrainedModel: ["qkv_net"],
+}
+
+# All the mapping for tensor parallelism
+TENSOR_PARALLEL_MAPPING = {
```

Comment:
I wrote the mappings of the two most representative models. Let's expand gradually.

Reply:
Let me add a mapping for t5:
```
'T5PreTrinedModel': {
        'column_parallel':['Attention.q', 'Attention.k', 'Attention.v', 'DenseReluDense.wi', 'relative_attention_bias'],
        'row_parallel':['Attention.o', 'DenseReluDense.wo'],
        'update_attrs':['n_heads', 'inner_dim']
  }
```
But the parameter `relative_attention_bias` is not in Conv1D layer or a nn.Linear layer, but its weight should changed in TP as well. Do we need create a new kind of mapping for it?

Reply:
Exceptions of this kind are increasing day by day. So I designed it like this.
https://github.com/tunib-ai/parallelformers/blob/main/parallelformers/policies/t5.py

Reply:
The most safe way is using an object that describe each module like this.
https://github.com/tunib-ai/parallelformers/blob/main/parallelformers/policies/base/policy.py#L23

Reply:
So, @lucasleesw As @stas00 said, Let's consider adding a dictionary for exception cases. Creating objects for every model is very tiring. Let's create a dictionary like ``LAYER REPLACEMENT`` and define there only for anomalous cases like T5. Does it make sense?

Reply:
Also some models will have an optional fused layernorm, e.g. very possibly with t5.

I've already mostly integrated apex.normalization.FusedRMSNorm in t5 https://github.com/huggingface/transformers/pull/14656 but for some reason it's not any faster, so need to find time to investigate why.

Reply:
@stas00 LayerNorm shouldn't be a problem since it doesn't affect tensor parallelization. Thanks for letting me know though.

Reply:
oh, cool, I misread then that somewhere in this PR you mentioned layernorm. but it was attention! 

I guess it'd be more important to know for kernel fusion discussion.

Reply:
I added T5 and `no_replacement` element like this. this is optional. does it make sense?

Reply:
```python
TENSOR_PARALLEL_MAPPING = {
    BertPreTrainedModel: {
        "col": ["query", "key", "value", "intermediate.dense"],
        "row": ["output.dense"],
        "update": ["num_attention_heads", "all_head_size"],
    },
    GPT2PreTrainedModel: {
        "col": ["c_attn", "q_attn", "c_fc"],
        "row": ["c_proj"],
        "update": ["embed_dim", "split_size", "num_heads"],
    },
    T5PreTrainedModel: {
        "col": ["Attention.q", "Attention.k", "Attention.v", "DenseReluDense.wi"],
        "row": ["Attention.o", "DenseReluDense.wo"],
        "row_no_replacement": ["relative_attention_bias"],
        "update": ["d_model", "n_heads", "inner_dim"],
    },
}
```